src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -1462,16 +1462,15 @@ void TurboAssembler::li(Register rd, Operand j, LiFlags mode) {
     }
 
     RecordRelocInfo(j.rmode(), immediate);
-    // FIXME(RISC_V): Does this case need to be constant size?
-    li_constant(rd, immediate);
+    li_ptr(rd, immediate);
   } else if (mode == ADDRESS_LOAD) {
     // We always need the same number of instructions as we may need to patch
-    // this code to load another value which may need all 8 instructions.
+    // this code to load another value which may need all 6 instructions.
     RecordRelocInfo(j.rmode());
-    li_constant(rd, j.immediate());
-  } else {  // mode == CONSTANT_SIZE - always emit the same instruction
+    li_ptr(rd, j.immediate());
+  } else {  // Always emit the same 48 bit instruction
```

Comment:
if `mode` is not `ADDRESS_LOAD`, then we need to use `li_constant` because it could be a full 64-bit value.

Reply:
The only places the size of instruction-sequence and their mnemonics are important are for creating addresses. I do not know any place that they generate 64 bit which is not an address (except in the two tests that are changed). Do you know of such cases. I put a DCHECK also that checks the upper 16 bit of the value to be zero in li_ptr(...) and the two routines retrieving and setting the address. I think li_constant(...) is not needed any longer (I know there are tests to test li_constant). As far as I know that is only used for pointers that li_ptr() replaced. More importantly there are cases that they call li_ptr( ) (previously li_constant) with a zero or a small number, to build a template for address initialization that later are filled with the actual address.

Reply:
Ok, I opened issue #274 to consolidate the two cases later.